benchmark-service/benchmark/benchmark.py

```diff
@@ -223,53 +225,118 @@ async def compare(request, userdata):  # pylint: disable=unused-argument
+
+    if has_results_file:
+        assert complete_batch_statuses, batch_statuses
+        log.info(f'commit {sha} has a results file')
+        status = complete_batch_statuses[-1]
+    elif running_batch_statuses:
+        status = running_batch_statuses[-1]
+        log.info(f'batch already exists for commit {sha}')
+    else:
+        status = None
+        log.info(f'no batches or results file exists for {sha}')
+
+    commit = {
+        'sha': sha,
+        'title': gh_commit['commit']['message'],
+        'author': gh_commit['commit']['author']['name'],
+        'date': gh_commit['commit']['author']['date'],
+        'status': status
     }
+    return commit
+
+
+async def update_commit(app, sha):  # pylint: disable=unused-argument
+    commit = await get_commit(app, sha)
+    if commit['status'] is not None:
+        return commit
+
+    batch_client = app['batch_client']
+
+    batch_id = await submit_test_batch(batch_client, sha)
+    batch = await batch_client.get_batch(batch_id)
+    commit['status'] = batch._last_known_status
+    log.info(f'submitted a batch {batch_id} for commit {sha}')
+
+    return commit
+
+
+@router.get('/api/v1alpha/benchmark/commit/{sha}')
+async def get_status(request):  # pylint: disable=unused-argument
+    sha = str(request.match_info['sha'])
+    app = request.app
+    commit = await get_commit(app, sha)
+    return web.json_response({'commit': commit})
+
+    # batch_client = app['batch_client']
+    # batch_statuses = [b._last_known_status async for b in batch_client.list_batches(q=f'sha={sha}')]
+    #
+    # try:
+    #     status = batch_statuses[-1]
+    #     log.info('got commit status')
+    #     return web.json_response({'status': status})
+    # except IndexError:
+    #     return web.json_response({'status': None})
+
+
+@router.delete('/api/v1alpha/benchmark/commit/{sha}')
+async def delete_commit(request):  # pylint: disable=unused-argument
+    global benchmark_data
+    app = request.app
+    gs_reader = app['gs_reader']
+    batch_client = app['batch_client']
+    sha = str(request.match_info['sha'])
+    file_path = f'{BENCHMARK_RESULTS_PATH}/{sha}.json'
+    if gs_reader.file_exists(file_path):
+        gs_reader.delete_file(file_path)
+        log.info(f'deleted file for sha {sha}')
+    async for b in batch_client.list_batches(q=f'sha={sha}'):
+        await b.delete()
+        log.info(f'deleted batch for sha {sha}')
+    if benchmark_data['commits'].get(sha):
+        del benchmark_data['commits'][sha]
+        log.info(f'deleted commit {sha} from commit list')
+    return web.Response()
+
+
+@router.post('/api/v1alpha/benchmark/commit/{sha}')
+async def call_update_commit(request):  # pylint: disable=unused-argument
+    body = await request.json()
+    sha = body['sha']
+    log.info('call_update_commit')
+    commit = await update_commit(request.app, sha)
+    return web.json_response({'commit': commit})
```

Comment:
Why does this need to be wrapped like this? Why not just `web.json_response(commit)`?